common/attest_plugin.c

```diff
@@ -11,63 +11,93 @@
 #include <stdio.h>
 #include <string.h>
 
+#include "attest_plugin.h"
 #include "common.h"
 
+#include <openenclave/attestation/verifier.h>
+#include <openenclave/bits/evidence.h>
+#include <openenclave/internal/plugin.h>
+
 const char* OE_REQUIRED_CLAIMS[OE_REQUIRED_CLAIMS_COUNT] = {
     OE_CLAIM_ID_VERSION,
     OE_CLAIM_SECURITY_VERSION,
     OE_CLAIM_ATTRIBUTES,
     OE_CLAIM_UNIQUE_ID,
     OE_CLAIM_SIGNER_ID,
     OE_CLAIM_PRODUCT_ID,
-    OE_CLAIM_PLUGIN_UUID};
+    OE_CLAIM_FORMAT_UUID};
 
 const char* OE_OPTIONAL_CLAIMS[OE_OPTIONAL_CLAIMS_COUNT] = {
     OE_CLAIM_VALIDITY_FROM,
     OE_CLAIM_VALIDITY_UNTIL};
 
-/**
- * Header that the OE runtime puts ontop of the attestation plugins.
- */
-typedef struct _oe_attestation_header
+// Variables storing the verifier list.
+static oe_plugin_list_node_t* verifiers = NULL;
+
+static void _print_hex_buffer_tail(
+    const char* title,
+    const uint8_t* buffer,
+    size_t size,
+    size_t tail)
 {
-    /* Set to OE_ATTESTATION_HEADER_VERSION. */
-    uint32_t version;
+    const size_t max_size = 64;
+    size_t offset = 0;
+    char* output = NULL;
 
-    /* UUID to identify format. */
-    oe_uuid_t format_id;
+    // Adjust for printing only the tail
+    if (tail && size > tail)
+    {
+        offset = size - tail;
+        size = tail;
+    }
 
-    /* Size of evidence/endorsements sent to the plugin. */
-    uint64_t data_size;
+    output = (char*)oe_malloc(max_size * 2 + 1);
+    if (!output)
+    {
+        OE_TRACE_ERROR("Out of memory for _print_hex_buffer()");
+        return;
+    }
 
-    /* The actual data */
-    uint8_t data[];
+    if (offset)
+        OE_TRACE_VERBOSE(
+            "%s[%d ->tail %d]:", title, (int)(size + offset), (int)size);
+    else
+        OE_TRACE_VERBOSE("%s[%d]:", title, (int)size);
 
-    /* data_size bytes that follows the header will be sent to a plugin. */
-} oe_attestation_header_t;
+    while (size > 0)
+    {
+        size_t segment_size = size;
+        if (segment_size > max_size)
+            segment_size = max_size;
+        oe_hex_string(
+            output, segment_size * 2 + 1, buffer + offset, segment_size);
+        output[segment_size * 2] = '\0';
+        OE_TRACE_VERBOSE("%s\n", output);
+        size -= segment_size;
+        offset += segment_size;
+    }
+    oe_free(output);
+}
 
-// Struct definition to represent the list of plugins.
-struct plugin_list_node_t
+static void _print_hex_buffer(
+    const char* title,
+    const uint8_t* buffer,
+    size_t size)
 {
-    oe_attestation_role_t* plugin;
-    struct plugin_list_node_t* next;
-};
-
-// Variables storing the attester and verifier lists.
-struct plugin_list_node_t* attesters = NULL;
-struct plugin_list_node_t* verifiers = NULL;
+    _print_hex_buffer_tail(title, buffer, size, 128);
```

Comment:
>128 [](start = 48, length = 3)

Why 128?

Reply:
_print_hex_buffer() is a helper debug function to log the tail of a large byte array when the SGX quote buffer was suspected to be corrupted. 128 was picked since only two lines would be logged but enough data to show if the buffer is corrupt or not.

Now that the debug is complete, maybe better to just delete the debug functions?